### DISCUSSION TITLE
Hook Method.invoke() crush

### DISCUSSION THREAD

**Discussion by fear1ess03** [LSPosed/LSPosed#1407] [Q&A]
when I hook Method.invoke()，the app will be suspend，so I think if method hook's implements uses Method.invoke() itself?

**Comment by yujincheng08** [comment]
yes

**Reply by fear1ess03** [reply]
so is there some method to resolve this problem?

**Reply by yujincheng08** [reply]
thats hard. invoking the original function is indeed calling invoke. so to prevent this, you need to
1. modify save the original native function pointer to executable.invoke, and then register another native function to call it, and let calling the original function always using your backuped function.
2. or, use native hook

**Reply by fear1ess03** [reply]
ok,how to use native hook?how can i find invoke's address?

**Reply by yujincheng08** [reply]
https://github.com/LSPosed/LSPosed/wiki/Native-Hook
